meps/mep-0003.md

```diff
@@ -0,0 +1,193 @@
+
+````
+```{tip}
+:class: dropdown
+:label: my-tip
+Content of the tip directive.
+```
+````
+
+or using the `---` syntax to write the options in YAML:
+
+````
+```{tip}
+---
+class: dropdown
+label: my-tip
+---
+Content of the tip directive.
+```
+````
+
+Each of the above is useful and will continue to be supported. For example, the multi-line syntax is helpful for long descriptions such as a caption. The YAML is useful when making those descriptions run over multiple lines themselves and take advantage of other YAML syntax for lists, etc. The single line syntax is much more concise.
+
+## UX implications & migration
+
+The proposed syntax is an addition to existing role and directive patterns. It is a stricly non-breaking enhancement; existing MyST documents that use this syntax are currently considered invalid and should not successfully parse. It does not invalidate or change the behavior of any existing workflows.
+
+The inline attribute syntax adopts the same inline attribute syntax used by Pandoc, Quarto, and djot, with the exception that the inline attribute must start with the role/directive name. This change follows from adapting the reference syntax to MyST Markdown, which already implements extensibilty through nominal types (directives and roles) rather than compositional types (i.e. class names).
+
+## Alternatives Considered
+
+We considered other approaches in this proposal and in discussion over several years. Including:
+
+1. extending the patterns used in Sphinx internal to the role content (`` {name}`content <argument .class1 key="value">` ``)
+2. attributes specified _after_ a role in addition to the name (``{name}`content`{.class1 key="value"}``); and
+3. bracketed spans `[content]{.class}`
+
+We chose the current approach to (a) separate the options and content clearly (i.e. not 1); (b) to keep role names and options close together (i.e. not 2); and \(c) to leave future syntax options open (i.e. 3 is a possible future addition).
+
+### Future Syntax Possibilities
+
+#### Parameterising Other Syntax Constructs
+
+This syntax is a stepping stone to other possible variants or additions to MyST, to be defined in future MEPs. For example, inline attributes for headers (e.g. `## My Header {#id .red}`), block attributes (e.g. preceding a block element with `{#id .red}=` (with or without the `=`), similar to the current `(label)=` syntax), or labelled spans/images/elements (e.g. `[content]{.red}` or `![](image.png){width="300px"}`).
+
+#### Short-hand Boolean Attibutes
+
+In certain markup languages, such as React JSX, it is possible to pass boolean `true` values simply by name, e.g.
+
+```
+<FooComponent myProp />
+```
+
+where `props.myProp` will be `true`. We might consider extending this syntax such that:
+
+````
+```{code .linenos}
+Numbered code directive
+
+Another line
+```
+````
+
+sets the `linenos` option to `true`.
```

Comment:
I like this, it is an interpretation of the class rather than new syntax.

Reply:
Oh, that might actually be a typo! I think `{code linenos}` is what I intended to write here.